cmd/krel/cmd/release_notes.go

```diff
@@ -113,12 +150,116 @@ func runReleaseNotes() (err error) {
 	logrus.Infof("Using start tag %v", start)
 	logrus.Infof("Using end tag %v", tag)
 
-	_, err = releaseNotesFrom(start)
+	if releaseNotesOpts.createDraftPR {
+		err = validateDraftPROptions()
+		if err != nil {
+			return errors.Wrap(err, "validating PR command line options")
+		}
+	}
+
+	result, err := releaseNotesFrom(start)
 	if err != nil {
 		return errors.Wrapf(err, "generating release notes")
 	}
 
-	//TODO: implement PR creation for k-sigs/release-notes and k/sig-release
+	// Create RN draft PR
+	if releaseNotesOpts.createDraftPR {
+		err = createDraftPR(tag, result)
+		if err != nil {
+			return errors.Wrap(err, "Failed to create release notes draft PR")
+		}
+	}
+
+	if releaseNotesOpts.outputDir != "" {
+		err = ioutil.WriteFile(releaseNotesOpts.outputDir+"release-notes.json", []byte(result.json), 0644)
+		if err != nil {
+			return errors.Wrap(err, "writing release notes JSON file")
+		}
+
+		err = ioutil.WriteFile(releaseNotesOpts.outputDir+"release-notes.md", []byte(result.json), 0644)
+		if err != nil {
+			return errors.Wrap(err, "writing release notes markdown file")
+		}
+	}
+
+	// TODO: implement PR creation for k-sigs/release-notes
+	return nil
+}
+
+// validateDraftPROptions checks if we have all needed parameters to create the Release Notes PR
+func validateDraftPROptions() error {
+	if releaseNotesOpts.createDraftPR {
+		// Check if --draft-org is set
+		if releaseNotesOpts.draftOrg == "" {
+			return errors.New("cannot generate Release Notes draft PR without --draft-org")
+		}
+
+		// Check if --draft-repo is set
+		if releaseNotesOpts.draftRepo == "" {
+			return errors.New("cannot generate Release Notes draft PR without --draft-repo")
+		}
```

Comment:
It's unfortunate that if you run it without either you'll get an error telling you to give `--draft-org`, then a second time you'll get an error telling you to give `--draft-repo`. As opposed to getting told you need both on the first invocation.

I don't think I have a better idea though. 🤔

Reply:
I've now added warning using logrus to advice the user is they are missing either